test/doctests.jl

```diff
@@ -0,0 +1,77 @@
+module DocTestsTest
+using Documenter
+using Compat.Test
+
+const bad_str = """
+```jldoctest
+julia> 1 + 2
+2
+
+julia> 1 + 1
+2
+```
+```jldoctest
+julia> 1 + 1
+2
+
+julia> 1 + 2
+2
+```
+```jldoctest
+julia> 1 + 2
+2
+
+julia> 1 + 2
+2
+```
+"""
+
+const good_str = """
+```jldoctest
+julia> 1 + 2
+3
+
+julia> 1 + 1
+2
+```
+```jldoctest
+julia> 1 + 1
+2
+
+julia> 1 + 2
+3
+```
+```jldoctest
+julia> 1 + 2
+3
+
+julia> 1 + 2
+3
+```
+"""
+
+println("="^50)
+@info("Testing doctest = :fixup")
+mktempdir(@__DIR__) do dir
+    @show dir
+    srcdir = mktempdir(dir)
+    @show srcdir
+    builddir = mktempdir(dir)
+    @show builddir
+    open(joinpath(srcdir, "index.md"), "w") do f
+        write(f, bad_str)
+    end
+    # fix up
+    makedocs(source = srcdir, build = builddir, doctest = :fixup)
+    # test that strict = true works
+    makedocs(source = srcdir, build = builddir, strict=true)
+    # also test that we obtain the expected output
+    open(joinpath(srcdir, "index.md"), "r") do f
+        str = read(f, String)
+        @test str == good_str
+    end
+end
```

Comment:
This should also fix doctests in `.jl` files too right? Could perhaps also test that to make sure that the file paths get resolved correctly etc.